app/assets/javascripts/components/option-select.js

```diff
@@ -56,21 +60,30 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     }
 
     // Attach listener to update checked count
-    this.$optionSelect.on('change', "input[type='checkbox']", this.updateCheckedCount.bind(this))
+    for (var j = 0; j < this.$options.length; j++) {
+      var anotherThis = this
+      this.$options[j].addEventListener('change', function () {
+        anotherThis.updateCheckedCount()
+      })
+    }
 
     // Replace div.container-head with a button
     this.replaceHeadingSpanWithButton()
 
     // Add js-collapsible class to parent for CSS
-    this.$optionSelect.addClass('js-collapsible')
+    this.$optionSelect.classList.add('js-collapsible')
 
     // Add open/close listeners
-    this.$optionSelect.find('.js-container-button').on('click', this.toggleOptionSelect.bind(this))
+    var button = this.$optionSelect.querySelector('.js-container-button')
+    var yetAnotherThis = this
+    button.addEventListener('click', function () {
+      yetAnotherThis.toggleOptionSelect()
```

Comment:
Thanks Andy 👍  I don't even pretend to understand `bind` - a `Hogwarts` method to me!  But, this may be my bad - sorry.  It looks like I'm not passing in `button` to the function and the `toggleOptionSelect()` call.  Once I updated this, I also found that the `if (e)` [here](https://github.com/alphagov/finder-frontend/pull/2525#discussion_r654320662) isn't needed!

Reply:
Great. I'd be tempted to checkout the original jQuery version and put a console log for `this` in that function, then compare the output with your code, just to check whether `this` is the same thing in both versions.

Reply:
Have put a `console.log` out for all occasions of `this` and `that` in all relevant functions and they all appear to point to the same `OptionSelect` element in both JQuery and Vanilla JS. This feels a bit weird as I thought they should reference different objects when the `this` reference is lost - hence the need to maintain that reference via a copy (`that`).  Testing for `==` and `===` sort of looks right, as in when you `var that = this` then immediately `console.log` them and compare with `==` or `===` they are the same, but inside a function (or event listener) they aren't.  I can't seem to find a solid method to compare the two objects (like you can in Ruby for example). So, still somewhat confused 🤯

Reply:
Finally managed to get `bind` working - still trying to understand it, but hey...!  One thing at a time right?